core/business.go

```diff
@@ -211,19 +211,20 @@ func NewWorker(wid int, jobPool chan chan Job) Worker {
 	return Worker{
 		Id:         wid,
 		JobPool:    jobPool,
-		JobChannel: make(chan Job),
+		JobChannel: make(chan Job, 5),
 		quit:       make(chan bool)}
 }
 
 // Start method starts the run loop for the worker, listening for a quit channel in
 // case we need to stop it
 func (w Worker) Start() {
 	var err error
+
+	// register the current worker into the worker queue.
+	w.JobPool <- w.JobChannel
```

Comment:
I want to understand if there is a side effect when using this outside of goroutine.